src/middleware/logger.rs

```diff
@@ -400,6 +421,7 @@ pub enum FormatText {
     RequestHeader(HeaderName),
     ResponseHeader(HeaderName),
     EnvironHeader(String),
+    CustomLog(String, Option<fn(req: &ServiceRequest) -> String>),
```

Comment:
since this is technically pub, we should indicate "request" like `CustomRequest`.